src/callbacks.jl

```diff
@@ -8,6 +8,25 @@
 # See https://github.com/jump-dev/JuMP.jl
 #############################################################################
 
+"""
+    callback_node_status(cb_data, model::Model)
+
+Return an [`MOI.CallbackNodeStatusCode`](@ref) enum, indicating if the current
+primal solution available from [`callback_value`](@ref) is integer feasible.
+"""
+function callback_node_status(cb_data, model::Model)
+    # TODO(odow):
+    # MOI defines `is_set_by_optimize(::CallbackNodeStatus) = true`.
+    # This causes problems for JuMP because it checks the termination_status to
+    # see if optimize! has been called. Solutions are:
+    # 1) defining is_set_by_optimize = false
+    # 2) adding a flag to JuMP to store whether it is in a callback
+    # 3) adding IN_OPTIMIZE to termination_status for callbacks
```

Comment:
I would vote for 3) as 1) would not work as the CachingOptimizer will try to query from the cache.
In addition, we could have `is_set_in_optimize` and the CachingOptimizer checks `is_set_in_optimizer || is_set_by_optimizer`.

Reply:
It might actually be easier just to have this slight work-around for callback attributes. We don't have many, and it's simpler than defining a new check with all the correct plumbing. If we have many more attributes, we can revisit this.